File: frontend/src/types/slm.ts
export interface ChatMessage {
  role: 'user' | 'assistant';
  content: string;
  timestamp?: string;
}

export interface ChatResponse {
  response: string;
  intent: string;
  confidence: number;
  model: string;
  suggestions: string[];
  timestamp: string;
  context_used: {
    current_price: number;
    net_position_oz: number;
    active_alerts: number;
  };
}

export interface SLMInfo {
  model_name: string;
  model_type: string;
  version: string;
  capabilities: string[];
  description: string;
}

File: backend/app/schemas/slm.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime


class ChatMessage(BaseModel):
    role: str  # "user" or "assistant"
    content: str
    timestamp: Optional[str] = None


class ChatRequest(BaseModel):
    message: str
    conversation_history: Optional[List[ChatMessage]] = None


class ChatResponse(BaseModel):
    response: str
    intent: str
    confidence: float
    model: str
    suggestions: List[str]
    timestamp: str
    context_used: Dict[str, Any]


class SLMInfoResponse(BaseModel):
    model_name: str
    model_type: str
    version: str
    capabilities: List[str]
    description: str
